src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -1176,6 +1181,42 @@ void TelemetryImpl::process_battery_status(const mavlink_message_t& message)
     }
 }
 
+void TelemetryImpl::process_battery_status_v2(const mavlink_message_t& message)
+{
+    mavlink_battery_status_v2_t bat_status;
+    mavlink_msg_battery_status_v2_decode(&message, &bat_status);
+
+    _has_bat_status = true;
+
+    Telemetry::Battery new_battery;
+    new_battery.id = bat_status.id;
+    new_battery.voltage_v = bat_status.voltage; 
```

Comment:
I think this does it right?

```suggestion
    new_battery.voltage_v = (std::numeric_limits<uint32_t>::max() == bat_status.voltage) ? NAN : bat_status.voltage * 1e-3f;
```

Reply:
Yes, that's good.